src/runtime.rs

```diff
@@ -0,0 +1,342 @@
+const _EXIT_ITERATIONS: &str = "_EXIT_ITERATIONS";
+const AWS_REGION: &str = "AWS_REGION";
+const RUNTIME_PATH: &str = "2018-06-01/runtime";
+const _X_AMZN_TRACE_ID: &str = "_X_AMZN_TRACE_ID";
+
+const TRACE_ID: HeaderName = HeaderName::from_static("lambda-runtime-trace-id");
+const DEADLINE_MS: HeaderName = HeaderName::from_static("lambda-runtime-deadline-ms");
+const REQUEST_ID: HeaderName = HeaderName::from_static("lambda-runtime-aws-request-id");
+const INVOKED_FUNCTION_ARN: HeaderName =
+    HeaderName::from_static("lambda-runtime-invoked-function-arn");
+const CLIENT_CONTEXT: HeaderName = HeaderName::from_static("lambda-runtime-client-context");
+const COGNITO_IDENTITY: HeaderName = HeaderName::from_static("lambda-runtime-cognito-identity");
+
+type HyperClient = Client<
+    hyper_rustls::HttpsConnector<hyper_util::client::legacy::connect::HttpConnector>,
+    Empty<Bytes>,
+>;
+
+#[derive(Clone)]
+struct LambdaContext<'js> {
+    pub aws_request_id: String,
+    pub invoked_function_arn: String,
+    pub log_group_name: String,
+    pub log_stream_name: String,
+    pub function_name: String,
+    pub function_version: String,
+    pub callback_waits_for_empty_event_loop: bool,
+    pub memory_limit_in_mb: usize,
+    pub get_remaining_time_in_millis: Function<'js>,
+    pub client_context: Value<'js>,
+    pub cognito_identity_json: Value<'js>,
+}
+
+struct NextInvocationResponse<'js> {
+    event: Value<'js>,
+    context: LambdaContext<'js>,
+}
+
+pub async fn runtime(ctx: &Ctx<'_>) -> Result<(), rquickjs::Error> {
+    static mut START_TIME: f64 = 0.0;
+
+    let aws_lambda_runtime_api = match get_env(AWS_LAMBDA_RUNTIME_API) {
+        Some(env) => env,
+        None => {
+            let msg = "Environment variable 'AWS_LAMBDA_RUNTIME_API' is not defined";
+            return Err(into_exception(msg.to_string(), ctx));
+        }
+    };
+
+    let (module_name, handler_name): (String, String) =
+        get_module_and_handler_name().map_err(|err| into_exception(err, ctx))?;
+    let task_root = get_task_root();
+
+    let js_handler_module: Object = Module::import(ctx, format!("{}/{}", task_root, module_name))?;
+    let js_init = js_handler_module.get::<_, Value>("init")?;
+    let js_bootstrap: Object = ctx.globals().get("__bootstrap")?;
+    let js_init_tasks: Array = js_bootstrap.get("initTasks")?;
+
+    if js_init.is_function() {
+        let idx = js_init_tasks.len();
+        let js_call: Object = js_init.as_function().unwrap().call(())?;
+        js_init_tasks.set(idx, js_call)?;
+    }
+
+    let init_tasks_sze = js_init_tasks.len();
+    ctx.globals().set("initTasks", js_init_tasks)?;
+
+    if init_tasks_sze == 1 {
+        ctx.eval::<(), _>("await initTasks[0]")?;
+    } else if init_tasks_sze > 1 {
+        ctx.eval::<(), _>("await Promise.all(initTasks)")?;
+    }
+
+    let handler: Value = js_handler_module.get(handler_name.as_str())?;
+
+    if !handler.is_function() {
+        panic!(
```

Comment:
We should not panic here but rather catch all errors that could occur in runtime and POST to:
`/init/error` with a json error body. See:
https://github.com/awslabs/llrt/blob/80b1cf91d467dc5ec4048f8f48bb4d764ad78faf/src/js/%40llrt/runtime.ts#L63